src/NHibernate/PropertyNotFoundException.cs

```diff
@@ -65,23 +59,41 @@ public PropertyNotFoundException(string propertyName, string fieldName, System.T
 		/// <param name="context">
 		/// The <see cref="StreamingContext"/> that contains contextual information about the source or destination.
 		/// </param>
-		protected PropertyNotFoundException(SerializationInfo info, StreamingContext context) : base(info, context)
+		protected PropertyNotFoundException(SerializationInfo info, StreamingContext context) 
+			: base(info, context)
 		{
+			foreach (var entry in info)
+			{
+				if (entry.Name == "PropertyName")
+				{
+					PropertyName = entry.Value?.ToString();
+				}
+				else if (entry.Name == "AccessorType")
+				{
+					AccessorType = entry.Value?.ToString();
+				}
+				else if (entry.Name == "TargetType")
+				{
+					var typeName = entry.Value?.ToString();
+					if (!string.IsNullOrEmpty(typeName))
+						TargetType = System.Type.GetType(typeName, true);
+				}
```

Comment:
When having many values, a `switch` seems more appropriate.
```c#
switch (entry.Name)
{
	case "PropertyName":
		PropertyName = entry.Value?.ToString();
		break;
	case "AccessorType":
		AccessorType= entry.Value?.ToString();
		break;
	case "TargetType":
		var typeName = entry.Value?.ToString();
		if (!string.IsNullOrEmpty(typeName))
			TargetType = System.Type.GetType(typeName, true);
		break;
}
```
(`DuplicateMappingException` and `LazyInitializationException` have two values, others only one.)

Reply:
I did consider this option and even written it this way, but for some reasons decided against it.